src/server/amqp.js

```diff
@@ -63,11 +64,23 @@ export function getNotifications(user, ws) {
             config.amqpExchangeName,
             NOTIFICATION_ROUTING_KEY + user
         );
-        logger.info("Websocket closed. Channel unbound for user: " + user);
+        msgChannel.cancel(consumerTag);
+        logger.info(
+            "Websocket closed. Channel unbound for user: " +
+                user +
+                " consumer Tag:" +
+                consumerTag
+        );
     });
 
     ws.on("error", function (error) {
         logger.info("Websocket error: " + error);
+        msgChannel.unbindQueue(
+            QUEUE,
+            config.amqpExchangeName,
+            NOTIFICATION_ROUTING_KEY + user
+        );
+        msgChannel.cancel(consumerTag);
         ws.close();
```

Comment:
Since we're calling close here, do we need to duplicate the unbind/cancel, or would it get handled in the close handler, above? I suppose it may not hurt to have both, as long as channel.cancel won't error out if the consumer is already canceled, though.

Reply:
Ah, good catch. I didn't notice that in my review, but it looks like it's now handled in the refactor in the last commit before this PR was merged (close and error are now calling a common cleanup function).

Reply:
Should I explicitly call `close` on websocket when there is an websocket error? I had that previously but removed it during refactor 🤔